### PR TITLE
Add configured defaults

### LINKED ISSUE #143: Lift configured_default behavior from CLI 2.0
This behavior is generic and useful to other Knack-based projects.

### PR DESCRIPTION
**Pull request by tjprescott**
cc/ @limingu

Closes #143.

Allows arguments to be backed by a value from the "defaults" section of the config file or an environment variable in the format `<CLI NAME>_DEFAULTS_<VAR NAME>`.

### CODE REVIEW

**Comment by limingu** [comment]
2 checks failed. Do you know why?

**Comment by tjprescott** [comment]
It was due to Python 2.7 having a different error message than 3.  It has been fixed.

File: tests/test_command_with_configured_defaults.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    @redirect_io
+    def test_no_configured_default_on_required_arg(self):
+        self._set_up_command_table(required=True)
+        with self.assertRaises(SystemExit):
+            self.cli_ctx.invoke('foo list'.split())
+        actual = self.io.getvalue()
+        expected = 'required: --my-param'
+        expected_2 = 'argument --my-param is required'  # Python 2.7
+        self.assertEqual(expected in actual or expected_2 in actual, True)
```
Comment (limingu) [inline tests/test_command_with_configured_defaults.py]: Will it be more clear and accurate to specify the exprected string as:
expected = 'required: --my-param' if sys.version_info[0] >= 3 else 'argument --my-param is required'

Comment (tjprescott) [reply tests/test_command_with_configured_defaults.py]: Done

**Pull request merged**